giskard/core/suite.py

```diff
@@ -76,10 +76,15 @@ def parse_function_arguments(client, project_key, function_inputs):
     return arguments
 
 
+TestName = str
+TestParams = Dict[str, Any]
+SuiteResult = Tuple[TestName, TestResult, TestParams]
```

Comment:
IMO this should be converted into a dataclass:

```python
@dataclass
def SuiteResult:
  test_name: str
  result: TestResult
  params: Dict[str, Any]
```

Reply:
I guess so, but the aliasing is making it clearer than it was still

Reply:
Good point, it might be used externally. However we can prevent backward compatibility issue with some kind of hack:

```python
def SuiteResult:
  test_name: str
  result: TestResult
  params: Dict[str, Any]
  
  @property
  def _tuple():
    # Method to ensure backward compatibility
    return (self.test_name, self.result, self.params)

  def __getitem__(self, i):
    warnings.warn("The test results has been changed to a class, getting attribute by index will be removed in the future.", category=DeprecationWarning)
    return self._tuple[i]

  def __iter__(self, i):
    warning.warn("The test results has been changed to a class, getting attribute by index will be removed in the future.", category=DeprecationWarning)
    return self._tuple.__iter__()
```

Reply:
That should do the trick. This behavior should work only for readers (as setter must now set the dataclass instead of the tuples).
I believe tests are extensively using tuple, maybe they should be some effort to clean those sometimes.

Reply:
Well tuple are immutable so we don't have to worry about writing

Reply:
Sorry, by writer I meant any code passing the test result as tuple must now pass the dataclass from now (again I don't know if it is somehow exposed externally, I modified it in testsuite and seems only internal)